sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -190,16 +192,27 @@ case class GpuWindowExpression(windowFunction: Expression, windowSpec: GpuWindow
       case other =>
         throw new IllegalStateException(s"${other.getClass} is not a supported window aggregation")
     }
+    // For Python UDF, it will be executed in Python processes. So return null here.
+    case _: GpuPythonUDF => null
     case other =>
       throw new IllegalStateException(s"${other.getClass} is not a supported window function")
   }
+  private lazy val windowInputProjection = if (windowFunc != null) {
+    windowFunc.windowInputProjection
+  }  else {
+    Seq.empty
+  }
   private lazy val boundRowProjectList = windowSpec.partitionSpec ++
-      windowFunc.windowInputProjection
+      windowInputProjection
   private lazy val boundRangeProjectList = windowSpec.partitionSpec ++
       windowSpec.orderSpec.map(_.child.asInstanceOf[GpuExpression]) ++
-      windowFunc.windowInputProjection
+      windowInputProjection
 
   override def columnarEval(cb: ColumnarBatch) : Any = {
+    if (windowFunc == null) {
```

Comment:
@mythrocks Do you have any suggestion for this ?

Reply:
Per my understanding, if it is a Python UDF, it should not go here, since Spark will create a separate physical plan for Python UDF.
Here is just a double check, i think.

Reply:
i think i found a better way, updated.